File: examples/php/check_balance.php
<?php
/**
 * MozeSMS API - Check Account Balance
 * 
 * Example showing how to check your account balance
 */

require_once 'config.php';

$config = include 'config.php';

/**
 * Get account balance
 */
function getBalance($config) {
    $url = $config['base_url'] . '/account/balance';
    
    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_HTTPHEADER, [
        'Content-Type: application/json',
        'X-API-Key: ' . $config['api_key'],
        'X-API-Secret: ' . $config['api_secret']
    ]);
    curl_setopt($ch, CURLOPT_TIMEOUT, $config['timeout']);
    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, $config['connect_timeout']);
    
    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    $error = curl_error($ch);
    curl_close($ch);
    
    if ($error) {
        return [
            'success' => false,
            'error' => 'cURL Error: ' . $error
        ];
    }
    
    $result = json_decode($response, true);
    
    return [
        'success' => $httpCode === 200,
        'http_code' => $httpCode,
        'data' => $result
    ];
}

// Example usage
echo "Checking account balance...\n\n";

$result = getBalance($config);

if ($result['success']) {
    echo "✅ Balance retrieved successfully!\n\n";
    echo "Current Balance: " . $result['data']['balance'] . " MZN\n";
    echo "SMS Credits: ~" . floor($result['data']['balance'] / 1.35) . " messages\n";
    echo "Currency: " . $result['data']['currency'] . "\n";
} else {
    echo "❌ Failed to get balance\n";
    echo "Error: " . ($result['error'] ?? $result['data']['message'] ?? 'Unknown error') . "\n";
}

echo "\nFull Response:\n";
print_r($result);

File: examples/python/check_balance.py
"""
MozeSMS API - Check Account Balance

Example showing how to check your account balance
"""

import requests
import config

def get_balance() -> dict:
    """
    Get account balance
    
    Returns:
        dict: Response data with balance information
    """
    url = f"{config.BASE_URL}/account/balance"
    
    headers = {
        'Content-Type': 'application/json',
        'X-API-Key': config.API_KEY,
        'X-API-Secret': config.API_SECRET
    }
    
    try:
        response = requests.get(
            url, 
            headers=headers,
            timeout=(config.CONNECT_TIMEOUT, config.TIMEOUT)
        )
        
        return {
            'success': response.status_code == 200,
            'http_code': response.status_code,
            'data': response.json()
        }
        
    except requests.exceptions.RequestException as e:
        return {
            'success': False,
            'error': str(e)
        }

if __name__ == '__main__':
    print('Checking account balance...\n')
    
    result = get_balance()
    
    if result['success']:
        print('✅ Balance retrieved successfully!\n')
        balance = result['data']['balance']
        currency = result['data']['currency']
        print(f"Current Balance: {balance} {currency}")
        print(f"SMS Credits: ~{int(balance / 1.35)} messages")
        print(f"Currency: {currency}")
    else:
        print('❌ Failed to get balance')
        error = result.get('error') or result.get('data', {}).get('message', 'Unknown error')
        print(f'Error: {error}')
    
    print('\nFull Response:')
    print(result)
